espresso/streamer.go

```diff
@@ -113,6 +113,15 @@ func NewEspressoStreamer[B Batch](
 func (s *EspressoStreamer[B]) Reset() {
 	s.BatchPos = s.fallbackBatchPos + 1
 	s.BatchBuffer.Clear()
+	// We have emptied our our batch buffer, so we've lost our place in what
+	// we were processing.  This is akin to starting over.  We don't currently
+	// have a known "good" starting position, so we just reset back to 0 for
+	// the hotshot position.
+	//
+	// TODO: This will need to be updated to fallback to a known good starting
+	//       position in the future. This is a temporary solution to ensure that
+	//       we will, at the very least, not stall and fail to make progress.
+	s.hotShotPos = 0
```

Comment:
Why not keep `s.hotShotPos = s.fallbackHotShotPos`. After Artemii's [update](https://github.com/EspressoSystems/optimism-espresso-integration/pull/119/files#diff-1912bc32db1ca7d02cb1784e7e053dc6d5a6e91043028c8160d48afaf23d2d3fR341-R360) isn't s.fallbackHotShotPos set correctly?

Reply:
Oh, this will be possible once you merge with celo-integration.

Reply:
Rebased from `celo-integration` and force pushed at 2d7e4a7a2bae6d62825e84e87611c5340a4246e7.